src/figuresClasses.ts

```diff
@@ -1,11 +1,115 @@
-export interface Figure {}
+export interface Figure {
+  color: string;
+  a?: number;
+  b?: number;
+  c?: number;
+  radius?: number;
+  width?: number;
+  height?: number;
+}
+
+export type FiguresData = {
+  shape: string;
+  color: string;
+  area: number;
+};
+
+export class Triangle implements Figure {
+  public shape = 'triangle';
+
+  public area = this.getArea();
+
+  constructor(
+    public color: string,
+    public a: number,
+    public b: number,
+    public c: number,
```

Comment:
[CODE KNOWLEDGE] - You don't need to manually assign constructor parameters to class properties when using TypeScript's parameter properties. The current implementation is redundant because TypeScript does this automatically when you use the 'public' keyword in the constructor parameters.